### PR TITLE
Implement SSK Sword

### PR DESCRIPTION
**Pull request by cpaca**
Currently untested; I'll test it another time. I'm making the PR now if you want to check the code, though.

### CODE REVIEW

File: src/main/java/xyz/holocons/mc/holoitemsrevamp/item/SSKSwordBook.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        if (enchantedMeta.addStoredEnchant(getEnchantment(), 1, false)) {
+            enchantedStack.setItemMeta(enchantedMeta);
+            enchantManager.removeCustomEnchantmentLore(enchantedStack);
+            enchantManager.applyCustomEnchantmentLore(enchantedStack);
+            return enchantedStack;
+        } else {
+            return null;
+        }
+    }
```
Comment (dlee13) [inline src/main/java/xyz/holocons/mc/holoitemsrevamp/item/SSKSwordBook.java:66]: This is wrong. EnchantmentStorageMeta is only for enchanted books

File: src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/SSKSword.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @Override
+    public void onPlayerAttack(EntityDamageByEntityEvent event, ItemStack itemStack) {
+        if(!(event.getEntity() instanceof LivingEntity target)) {
+            return;
+        }
+
+        // This is what Klin used, and it seems to be equivalent to damage after-sharpness after-plugins before-prot4
+        final var amountToHeal = event.getDamage();
+        Util.healEntity(target, amountToHeal);
+        event.setDamage(0);
```
Comment (dlee13) [inline src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/SSKSword.java:71]: I think you could just set the damage to negative to heal. Gotta test that.

File: src/main/java/xyz/holocons/mc/holoitemsrevamp/item/SSKSwordBook.java (hunk 1 above)
Comment (dlee13) [reply src/main/java/xyz/holocons/mc/holoitemsrevamp/item/SSKSwordBook.java:66]: Wait, why is this an enchanted book?

**Comment by cpaca** [comment]
Just tested; the main functionality seems to work. Still need to test other stuff (recipes, setting damage to negative, etc)

File: src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/SSKSword.java (hunk 2 above)
Comment (cpaca) [reply src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/SSKSword.java]: Yep, it works. I'm gonna keep healEntity in Util and port it to DAura at some point.

**Comment by cpaca** [comment]
Tested the recipe. 

Negative damage is overhealing. Will ask about that more in discord.

**Comment by dlee13** [comment]
So you know, Bukkit#addRecipe already registers the flipped recipes. There's no reason to manually reverse them like Klin did. And you don't need to put comments about left/right recipes or recipe0 or whatever.

**Comment by cpaca** [comment]
Oh, 1.21 has a heal api method? I might pull #29 into draft then just so it can use that, even if it only really changes one line of code.
